### PR TITLE
Fix bandd init with custom parameter

### LINKED ISSUE #624: Finalize wenchange chain parameters
Study the parameters of existing chains especially Cosmos, Kava, and Terra. Discuss with @sorawit and finalize the parameters. Once this is final, put it in `bandd/main.go` so `bandd init` automatically includes that configuration.

### PR DESCRIPTION
**Pull request by taobun**
fixed #624 

[Parameter summary](https://docs.google.com/spreadsheets/d/1-BpqJPPLYoqIQJFHmTEskX4HXFfS61b98pDRd1OmBt4/edit?usp=sharing)

### CODE REVIEW

File: chain/app/genesis.go [hunk 1]
```diff
@@ -1,11 +1,120 @@
...
+				VotingPeriod: 86400 * 2 * time.Second, // 2 days
+			},
+			TallyParams: gov.TallyParams{
+				Quorum:    sdk.NewDecWithPrec(4, 1),
+				Threshold: sdk.NewDecWithPrec(5, 1),
+				Veto:      sdk.NewDecWithPrec(334, 3),
+			},
+		}),
+		crisis.ModuleName: crisis.ModuleCdc.MustMarshalJSON(crisis.GenesisState{
+			ConstantFee: sdk.NewCoin("uband", sdk.NewInt(1000)),
```
Comment (sorawit) [inline chain/app/genesis.go]: Doesn't seem right.

File: chain/app/genesis.go [hunk 2]
```diff
@@ -1,11 +1,120 @@
...
+			StartingProposalID: 1,
+			DepositParams: gov.DepositParams{
+				MinDeposit:       sdk.Coins{sdk.NewCoin("uband", sdk.TokensFromConsensusPower(1000))},
+				MaxDepositPeriod: 86400 * 2 * time.Second, // 2 days
+			},
+			VotingParams: gov.VotingParams{
+				VotingPeriod: 86400 * 2 * time.Second, // 2 days
+			},
+			TallyParams: gov.TallyParams{
+				Quorum:    sdk.NewDecWithPrec(4, 1),
```
Comment (sorawit) [inline chain/app/genesis.go]: Add comment please. // 0.4 // 0.5 // 0.334

File: chain/app/genesis.go [hunk 3]
```diff
@@ -1,11 +1,120 @@
...
+				MaxValidators: 100,
+			},
+		}),
+		mint.ModuleName: mint.ModuleCdc.MustMarshalJSON(mint.GenesisState{
+			Minter: mint.Minter{
+				AnnualProvisions: sdk.NewDecWithPrec(0, 0),
+				Inflation:        sdk.NewDecWithPrec(135, 3),
+			},
+			Params: mint.Params{
+				BlocksPerYear:       10519200, //  assuming 3 second  block times
```
Comment (sorawit) [inline chain/app/genesis.go]: 3-second block time

File: chain/app/genesis.go [hunk 4]
```diff
@@ -1,11 +1,120 @@
...
+					SigVerifyCostED25519:   590,
+					SigVerifyCostSecp256k1: 1000,
+				},
+			}),
+		bank.ModuleName: bank.ModuleCdc.MustMarshalJSON(bank.GenesisState{
+			SendEnabled: true,
+		}),
+		staking.ModuleName: staking.ModuleCdc.MustMarshalJSON(staking.GenesisState{
+			Params: staking.Params{
+				UnbondingTime: time.Hour * 24 * 7 * 3,
```
Comment (sorawit) [inline chain/app/genesis.go]: // 3 weeks

File: chain/app/genesis.go [hunk 5]
```diff
@@ -1,11 +1,120 @@
...
+				Threshold: sdk.NewDecWithPrec(5, 1),
+				Veto:      sdk.NewDecWithPrec(334, 3),
+			},
+		}),
+		crisis.ModuleName: crisis.ModuleCdc.MustMarshalJSON(crisis.GenesisState{
+			ConstantFee: sdk.NewCoin("uband", sdk.NewInt(1000)),
+		}),
+		slashing.ModuleName: slashing.ModuleCdc.MustMarshalJSON(slashing.GenesisState{
+			Params: slashing.Params{
+				MaxEvidenceAge:          60 * 30240 * time.Second,
```
Comment (sorawit) [inline chain/app/genesis.go]: time.Hour * 24 * 7 * 3 // 3 weeks

File: chain/cmd/bandd/main.go [hunk 6]
```diff
@@ -64,10 +71,122 @@ func GenFilePVIfNotExists(cdc *codec.Codec, keyFilePath, stateFilePath string) {
...
+			genDoc.Validators = nil
+			genDoc.AppState = appState
+			genDoc.ConsensusParams = &types.ConsensusParams{
+				Block: types.BlockParams{
+					MaxBytes:   200000, // 0.2MB
+					MaxGas:     2000000,
+					TimeIotaMs: 1000, // 1s
+				},
+				Evidence: types.EvidenceParams{
+					MaxAge: 100000, // 27.8 hrs at 1block/s
```
Comment (sorawit) [inline chain/cmd/bandd/main.go]: Don't think this comment is accurate, but we can just remove it honestly.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/cmd/bandd/main.go [hunk 7]
```diff
@@ -64,10 +71,122 @@ func GenFilePVIfNotExists(cdc *codec.Codec, keyFilePath, stateFilePath string) {
...
+		return err
+	}
+
+	_, err = fmt.Fprintf(os.Stderr, "%s\n", string(sdk.MustSortJSON(out)))
+	return err
+}
+
+// InitCmd returns a command that initializes all files needed for Tendermint
+// and the respective application.
+func InitCmd(ctx *server.Context, cdc *codec.Codec, customAppState map[string]json.RawMessage,
```
Comment (sorawit) [inline chain/cmd/bandd/main.go]: Also I think it makes sense to move this to its own file.

File: chain/cmd/bandd/main.go [hunk 8]
```diff
@@ -64,10 +71,122 @@ func GenFilePVIfNotExists(cdc *codec.Codec, keyFilePath, stateFilePath string) {
...
+
+	cmd.Flags().String(cli.HomeFlag, defaultNodeHome, "node's home directory")
+	cmd.Flags().BoolP(flagOverwrite, "o", false, "overwrite the genesis.json file")
+	cmd.Flags().String(client.FlagChainID, "", "genesis file chain-id, if left blank will be randomly created")
+
+	return cmd
+}
+
 func main() {
 	cobra.EnableCommandSorting = false
```
Comment (sorawit) [inline chain/cmd/bandd/main.go]: Now that we write our own InitCmd. Can you also remove my hacky `PersistentPreRunE`? Basically revert my commit 7b41ef74081601ea0f3d9fb0930a2a13b4ee8287

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**